### ISSUE TITLE
Unable to Switch to WorldVoice when IBM is Initially Enable

### ISSUE THREAD

**Issue by lbk2907** [tsengwoody/WorldVoice#39]
### A Brief Description
When attempting to activate IBMTTS in WorldVoice, NVDA fails to switch synthesizers and falls back to previous synthesizer. The error occurs when the rateBoost setting hasn't been previously configured with SAPI5 voices.

### Error Log
```
INFO - NVDAHelperLocal (09:29:30.843) - MainThread (15308):
Thread 15308, build\x86\localWin10\oneCoreSpeech.cpp, ocSpeech_initialize, 215:
ocSpeech_initialize

INFO - NVDAHelperLocal (09:29:30.843) - MainThread (15308):
Thread 15308, build\x86\localWin10\oneCoreSpeech.cpp, OcSpeechState::activate, 89:
Activating

INFO - NVDAHelperLocal (09:29:30.864) - MainThread (15308):
Thread 15308, build\x86\localWin10\oneCoreSpeech.cpp, preventEndUtteranceSilence_, 443:
AppendedSilence supported

ERROR - synthDriverHandler.setSynth (09:29:31.133) - MainThread (15308):
setSynth failed for WorldVoice
Traceback (most recent call last):
  File "synthDriverHandler.pyc", line 487, in setSynth
  File "synthDriverHandler.pyc", line 454, in getSynthInstance
  File "synthDriverHandler.pyc", line 343, in initSettings
  File "C:\Users\Calvin\AppData\Roaming\nvda\addons\WorldVoice\synthDrivers\WorldVoice\__init__.py", line 367, in loadSettings
    super().loadSettings(*args, **kwargs)
  File "synthDriverHandler.pyc", line 362, in loadSettings
  File "config\__init__.pyc", line 1165, in __getitem__
KeyError: 'rateBoost'
INFO - synthDriverHandler.setSynth (09:29:31.134) - MainThread (15308):
Falling back to previous synthDriver sapi5
INFO - synthDriverHandler.setSynth (09:29:31.296) - MainThread (15308):
Loaded synthDriver sapi5
```

### Steps to Reproduce
1. Start with fresh NVDA settings (or ensure rateBoost hasn't been configured for SAPI5 in WorldVoice).
2. Open WorldVoice Settings.
3. Go to Speech Engine tab.
4. Enable IBM and press OK.
5. Place IBMTTS voice data in "%appdata%\nvda\WorldVoice-workspace\IBM".
6. Restart NVDA.
7. Attempt to switch to WorldVoice synthesizer.

### Expected Result
WorldVoice should activate successfully.
Actual Result:
NVDA fails to switch to WorldVoice and falls back to previous synthesizer.

### Workaround
1. Make sure IBM is not enable or the IBM voice data isn't in the spesified path (%appdata%\nvda\WorldVoice-workspace\IBM).
2. Switch to SAPI5 voice in WorldVoice.
3. Toggle the rate boost setting on/off.
4. Enable IBM or put the IBM voice data into the spesified path (%appdata%\nvda\WorldVoice-workspace\IBM).
5. Switch to another synthesizer.
6. Save the configuration.
7. Restart NVDA.
8. Attempt to switch to WorldVoice.

### System Information
WorldVoice version: 5.0.
NVDA version: 2025.1.1 (2025.1.1.36849).
• OS: Windows 11 24H2 (AMD64) build 26100.4351.

### Additional Notes
The default inflection value is set to 50 and cannot be adjusted when using IBM. Inflection should be adjustable (typically between 0 to 100). Currently, no known method exists to modify this setting.

**Comment by tsengwoody** [comment]
> Start with fresh NVDA settings (or ensure rateBoost hasn't been configured for SAPI5 in WorldVoice).
Please provide the configuration file (nvda.ini) that reproduces this status.

**Comment by lbk2907** [comment]
Which status? Are the status which can't run the IBM or the one that can
run IBM? If the configuration that cannot run IBM, just restore NVDA
configuration to default.

**Comment by tsengwoody** [comment]
I can’t reproduce this error. I pressed “NVDA+Ctrl+R” three times to reset the settings to default, then went to the speech engine settings to enable the IBM engine, but the error did not occur. Maybe there are other hidden conditions required to trigger this issue.

**Comment by lbk2907** [comment]
I not really sure. It happend on me. May I know where you put the IBM data
and what you get in the IBM folder? Mine is put in the WorldVoice-workspace
folder called "IBM". In that folder is the IBM data. Mine is the one that
have Chinese Japanese Korean (CJK). May I know what can I help you to find
out this issue? O yeah, don't forget that this error only happend when you
have the IBM data in the folder mentioned earlier. If you need the IBM data
for the debugging purpose, field free to contact me privately. You can find
my contact information in my profile.

**Comment by tsengwoody** [comment]
Fix this issue in commit 3766bb.

**Comment by tsengwoody** [comment]
Hi,
This issue has been resolved as part of an internal refactor of WorldVoice.
The related code has already been removed in the current version.
Thanks for reporting!
